include/flatbuffers/flatbuffers.h

```diff
@@ -960,9 +960,11 @@ FLATBUFFERS_FINAL_CLASS
   }
 
   /// @brief Store a string in the buffer, which can contain any binary data.
-  /// @param[in] str A const reference to a std::string to store in the buffer.
+  /// @param[in] str A const reference to a std::string like type with support
+  /// of T::c_str and T::length() to store in the buffer.
   /// @return Returns the offset in the buffer where the string starts.
-  Offset<String> CreateString(const std::string &str) {
+  template<typename T>
+  Offset<String> CreateString(const T &str) {
```

Comment:
@aardappel  Yeah, here is a test for you: 

http://cpp.sh/77moa

Given this test, I don't think we need a std::string overload. 

I'll add it, but it seems like dead code to me. 

Let me know.